src/externalized/DefaultContentManager.cc

```diff
@@ -1185,18 +1193,33 @@ bool DefaultContentManager::loadMercsData()
 	}
 	MercProfileInfo::validateData(m_mercProfileInfo);
 
-	json = readJsonDataFileWithSchema("mercs-relations.json");
-	for (auto& element : json.toVec()) {
-		JsonObject reader = element.toObject();
-		const MercProfileInfo* inf = this->getMercProfileInfoByName(reader.GetString("profile"));
-		MercProfile::deserializeStructRelations(temp_mercStructs[inf->profileID].get(), reader, this);
-	}
-	for (auto& element : temp_mercStructs) {
-		if (element != nullptr) {
-			m_mercStructs.push_back( std::make_unique<const MERCPROFILESTRUCT>( *element ) );
+	if (!doesGameResBelongToMod(BINARYDATADIR "/prof.dat")) {
+		json = readJsonDataFileWithSchema("mercs-relations.json");
+		for (auto& element : json.toVec()) {
+			JsonObject reader = element.toObject();
+			const MercProfileInfo* inf = this->getMercProfileInfoByName(reader.GetString("profile"));
+			MercProfile::deserializeStructRelations(temp_mercStructs[inf->profileID].get(), reader, this);
 		}
-		else {
-			m_mercStructs.push_back( std::make_unique<const MERCPROFILESTRUCT>() );
+		for (auto& element : temp_mercStructs) {
+			if (element != nullptr) {
+				m_mercStructs.push_back(std::make_unique<const MERCPROFILESTRUCT>(*element));
+			}
+			else {
+				m_mercStructs.push_back(std::make_unique<const MERCPROFILESTRUCT>());
+			}
+		}
+	}
+	else {
+		AutoSGPFile f(openGameResForReading(BINARYDATADIR "/prof.dat"));
+		bool const isCorrectlyEncoded = !(isRussianVersion() || isRussianGoldVersion());
+		for (int i = 0; i != NUM_PROFILES; ++i) {
+			BYTE data[MERC_PROFILE_SIZE];
+			JA2EncryptedFileRead(f, data, sizeof(data));
+			std::unique_ptr<MERCPROFILESTRUCT> prof = std::make_unique<MERCPROFILESTRUCT>();
+			UINT32 checksum;
+			ExtractMercProfile(data, *prof, false, &checksum, isCorrectlyEncoded);
+			// not checking the checksum
+			m_mercStructs.push_back(std::make_unique<const MERCPROFILESTRUCT>(*prof));
```

Comment:
No need to make a copy here, `m_mercStructs.push_back(std::move(prof));` should suffice.